static/js/server_events_dispatch.js

```diff
@@ -409,6 +409,7 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
         }
         if (event.setting_name === 'demote_inactive_streams') {
             stream_list.update_streams_sidebar();
+            stream_data.set_filter_out_inactives();
```

Comment:
Shouldn't we add a call to this whenever the user is subscribed/unsubscribed from a stream as well?  

I think that would avoid having a real-time sync bug.

Reply:
It's a judgment call what's the right behavior here.  If somebody subscribes me to a stream, do we want to gray out streams without the user understanding what's happening?

Reply:
Yeah, I think it's reasonable to make the settings determination once on page load (and again if you change that setting).  OK.